cloudify_agent/worker.py

```diff
@@ -176,15 +194,73 @@ def new_excepthook(exception_type, value, the_traceback):
     sys.excepthook = new_excepthook
 
 
+class _ProcessRegistry(object):
+    """A registry for dispatch subprocesses.
+
+    The dispatch TaskHandler uses this to register the subprocesses that
+    are running and executing a task, so that they can be cancelled/killed
+    from outside.
+    """
+    def __init__(self):
+        self._processes = {}
+
+    def register(self, handler, process):
+        self._processes[self.make_key(handler)] = process
+
+    def unregister(self, handler):
+        self._processes.pop(self.make_key(handler), None)
+
+    def cancel(self, operation_id):
+        p = self._processes.get(operation_id)
+        if p:
+            t = threading.Thread(target=self._stop_process, args=(p, ))
+            t.start()
+
+    def _stop_process(self, process):
```

Comment:
I guess if you have class A and classes B and C inherit from it, and you only ever use B and C, and C defines method `foo`, then I guess `foo` should only exist in B and not A.
This is a roundabout way to say - let's maybe move `_stop_process` to `OperationProcessRegistry`

Reply:
nah C uses it as well (calls super, right)